### PR TITLE
Refactor ML section - local and remote models

### LINKED ISSUE #4866: [DOC] Neural Search Plugin Needs a Prerequisites Section and General Architecture
We should add a prerequisites section to the Neural Search page. This section should cover that you need either a deployed model using ML Commons (local) or a ML Connector (remote inference).  

https://opensearch.org/docs/latest/search-plugins/neural-search/
 
- [X] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 There prerequisite steps to using the Neural Search plugin. You need a model_id to create the index and need to know the dimensionality of the model. Additionally it would help to have a general architecture diagram to understand how the components work together. 

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
N/A

### LINKED ISSUE #4966: [DOC] ml-framework page example request doesn't work
**What do you want to do?**
 
- [X] Request a change to existing documentation

**Tell us about your request.** Provide a summary of the request and all versions that are affected.

https://opensearch.org/docs/latest/ml-commons-plugin/ml-framework/ - 
The example request here doesn't seem to work. It's missing the `model_content_hash_value` key.

image

Regarding that, it's not mentioned anywhere. When registering a model via URL, you'll have to pass that value and it's the sha256 checksum of the model zipfile. You'll want to provide instructions on how to calculate that and make it part of the call. 

For comparison, the example call to register a model from the API reference page for machine learning works proper-like. 

https://opensearch.org/docs/latest/ml-commons-plugin/api/

image

### LINKED ISSUE #5559: [DOC] Supported model_format in ML Framework
**What do you want to do?**
 
- [X] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

[Here](https://opensearch.org/docs/latest/ml-commons-plugin/ml-framework/#model-format) both ONNX and TorchScript formats are supported but in the [request fields table](https://opensearch.org/docs/latest/ml-commons-plugin/ml-framework/#upload-model-to-opensearch), the `model_format` parameter only supports `TORCH_SCRIPT`.
Which formats are supported?

### LINKED ISSUE #5591: [DOC] Fix wrong example
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 https://opensearch.org/docs/latest/ml-commons-plugin/ml-framework/#example-request

MIssed `model_content_hash_value` , refer to https://github.com/opensearch-project/documentation-website/issues/4966

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by kolchfa-aws**
### Description
Refactor ML section - local and remote models

### Issues Resolved
Fixes #5591 
Fixes #4866 
Fixes #4966
Fixes #5559

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _ml-commons-plugin/api/model-apis/index.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+has_children: true
+nav_order: 10
+---
+
+# Model APIs
+
+ML Commons supports the following model-level APIs:
+
+- [Register model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/register-model/)
+- [Deploy model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/deploy-model/)
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/api/model-apis/index.md:14]: minor: move this line after Get model?

File: _ml-commons-plugin/api/model-apis/index.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+- [Get model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/get-model/)
+- [Undeploy model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/undeploy-model/)
+- [Delete model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/delete-model/)
+
+## Model access control considerations
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/api/model-apis/index.md:19]: This is mainly about model groups, Should we move this to `model-group-apis/index.md` ?

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+## Path and HTTP methods
+
+```json
+POST /_plugins/_ml/models/_register
+```
+{% include copy-curl.html %}
+
+## Request fields
+
+All request fields are required.
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/api/model-apis/register-model.md:36]: This is not correct.
For different model, they need different fields. 

For pretrained model, check https://opensearch.org/docs/latest/ml-commons-plugin/pretrained-models/, the mandatory fields are `name`, `version`, `model_format`

For uploading model via URL, refer to https://github.com/opensearch-project/ml-commons/blob/2.x/docs/model_serving_framework/text_embedding_model_examples.md, mandatory fields:  

- `name`
- `version`
- `model_task_type`
- `model_format`
- `model_content_hash_value`
- `model_config`: {`model_type`, `embedding_dimension`, `framework_type`}, 
- `url`

Comment (ylwu-amzn) [reply _ml-commons-plugin/api/model-apis/register-model.md:36]: For remote model , refer to https://github.com/opensearch-project/ml-commons/blob/2.x/docs/tutorials/remote_inference.md

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  "worker_node": [
+    "N77RInqjTSq_UaLh1k0BUg"
+  ],
+  "create_time": 1685478486057,
+  "last_update_time": 1685478491090,
+  "is_async": true
+}
+```
+
+## Registering a model containing an internal connector
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/api/model-apis/register-model.md:126]: Refer to https://github.com/opensearch-project/ml-commons/blob/2.x/docs/tutorials/remote_inference.md

For Register remote model, user can use internal connector or connector id (standalone connector)

File: _ml-commons-plugin/api/model-apis/index.md (hunk 2 above)
Comment (kolchfa-aws) [reply _ml-commons-plugin/api/model-apis/index.md:19]: This is about model groups but from the perspective of accessing a model. This section gives a brief overview of what models can be accessed by what users and then links to the model access control page for further details. So I think I'll leave this here.

File: _ml-commons-plugin/cluster-settings.md [hunk 5]
```diff
@@ -19,7 +19,11 @@ node.roles: [ ml ]
 
 ## Run tasks and models on ML nodes only
 
-If `true`, ML Commons tasks and models run machine learning (ML) tasks on ML nodes only. If `false`, tasks and models run on ML nodes first. If no ML nodes exist, tasks and models run on data nodes. We recommend that you do not set this value to "false" on production clusters. 
+If `true`, ML Commons tasks and models run machine learning (ML) tasks on ML nodes only. If `false`, tasks and models run on ML nodes first. If no ML nodes exist, tasks and models run on data nodes.
```
Comment (vagimeli) [inline _ml-commons-plugin/cluster-settings.md:22]:
```suggestion
If `true`, ML Commons tasks and models run ML tasks on ML nodes only. If `false`, tasks and models run on ML nodes first. If no ML nodes exist, tasks and models run on data nodes. 
```

File: _ml-commons-plugin/custom-local-models.md [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+  "function_name": "REMOTE",
+  "state": "COMPLETED",
+  "worker_node": [
+    "n-72khvBTBi3bnIIR8FTTw"
+  ],
+  "create_time": 1689793851077,
+  "last_update_time": 1689793851101,
+  "is_async": true
+}
+```
```
Comment (vagimeli) [inline _ml-commons-plugin/custom-local-models.md:208]: Do we need a blank line between lines 208 and 209?

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+| `framework_type` | String  | The framework the model is using. Currently, we support `sentence_transformers` and `huggingface_transformers` frameworks. The `sentence_transformers` model outputs text embeddings directly, so ML Commons does not perform any post processing. For `huggingface_transformers`, ML Commons performs post processing by applying mean pooling to get text embeddings. See the example [`all-MiniLM-L6-v2` Hugging Face model](https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2) for more details. Required. |
+| `all_config` | String | This field is used for reference purposes. You can specify all model configurations in this field. For example, if you are using a Hugging Face model, you can minify the `config.json` file to one line and save its contents in the `all_config` field. Once the model is uploaded, you can use the get model API operation to get all model configurations stored in this field. Optional. |
```
Comment (vagimeli) [inline _ml-commons-plugin/api/model-apis/register-model.md:131]: Should this read "Get Model API?"

File: _ml-commons-plugin/api/profile.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
+---
+layout: default
+title: Profile
+parent: ML Commons API
+nav_order: 40
+---
+
+# Profile
+
+The profile API operation returns runtime information about ML tasks and models. The profile operation can help debug model issues at runtime.
```
Comment (vagimeli) [inline _ml-commons-plugin/api/profile.md:10]: Should this read "Profile API operation?"

File: _ml-commons-plugin/api/train-predict/train.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+title: Train 
+parent: Train and Predict APIs
+grand_parent: ML Commons API
+has_children: true
+nav_order: 10
+---
+
+# Train 
+
+The train API operation trains a model based on a selected algorithm. Training can occur both synchronously and asynchronously.
```
Comment (vagimeli) [inline _ml-commons-plugin/api/train-predict/train.md:12]: Should this read "Train API operation?"

**Review by vagimeli — APPROVED** [review]
@kolchfa-aws Solid documentation. Edits are minimal. I agree with capitalizing the API sections because they serve as headings.

File: _ml-commons-plugin/api/model-apis/register-model.md [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+    "model_group_id": "Z1eQf4oB5Vm0Tdw8EIP2",
+    "description": "This is a neural sparse encoding model: It transfers text into sparse vector, and then extract nonzero index and value to entry and weights. It serves only in ingestion and customer should use tokenizer model in query.",
+    "model_format": "TORCH_SCRIPT",
+    "function_name": "SPARSE_ENCODING",
+    "model_content_hash_value": "9a41adb6c13cf49a7e3eff91aef62ed5035487a6eca99c996156d25be2800a9a",
+    "url":  "https://artifacts.opensearch.org/models/ml-models/amazon/neural-sparse/opensearch-neural-sparse-encoding-doc-v1/1.0.0/torch_script/opensearch-neural-sparse-encoding-doc-v1-1.0.0-torch_script.zip"
+}
+```
+{% include copy-curl.html %}
+
```
Comment (kolchfa-aws) [inline _ml-commons-plugin/api/model-apis/register-model.md:102]:
```suggestion
```

File: _ml-commons-plugin/custom-local-models.md (hunk 6 above)
Comment (kolchfa-aws) [inline _ml-commons-plugin/custom-local-models.md:208]:
````suggestion
```

````

**Pull request merged**